met-web/src/components/publicDashboard/SubmissionTrend/SubmissionTrend.tsx

```diff
@@ -46,14 +67,30 @@ const SubmissionTrend = ({ engagement, engagementIsLoading }: SubmissionTrendPro
     };
 
     useEffect(() => {
-        fetchData();
+        fetchData().catch((error) => {
+            console.error(error);
+        });
     }, [engagement.id, chartBy]);
 
+    useEffect(() => {
```

Comment:
do we have to use two useEffects to call the same function, can we do better?

Reply:
I think if we just add the dependencies [fromDate, toDate] to the first one it should work, can you please try that?

Reply:
done please review again